server/data_common/driver.py

```diff
@@ -0,0 +1,319 @@
+from abc import ABCMeta, abstractmethod
+from server_timing import Timing as ServerTiming
+import numpy as np
+import pandas as pd
+from server.data_common.fbs.matrix import encode_matrix_fbs
+from server.common.constants import Axis, DEFAULT_TOP_N
+from server.common.errors import FilterError, JSONEncodingValueError
+from server.compute.diffexp import diffexp_ttest
+from server.data_common.utils import jsonify_numpy
+from server.data_common.matrix_proxy import MatrixProxy
+from server.common.app_config import AppFeature, AppSingleConfig
+
+"""
+Sort order for methods
+1. Initialize
+2. Helper
+3. Filter
+4. Data & Metadata
+5. Computation
+"""
+
+
+class CXGDriver(metaclass=ABCMeta):
```

Comment:
i find the mixed terminology of "Driver" vs "Engine" confusing.  Any reason we can't pick one?

For example:
* "AbstractEngine" and "TileDbEngine", or
* "DataAdaptor" and "TileDbDataAdaptor"

I kind of like naming it what it is -- a data adaptor and/or plugin

Reply:
I renamed everything to "Adaptor"